tests/parametric/test_span_links.py

```diff
@@ -0,0 +1,189 @@
+        assert (link.get("flags") or 0) == 0
+
+    def test_span_link_from_distributed_datadog_headers(self, test_agent, test_library):
+        """Properly inject datadog distributed tracing information into span links.
+        Testing the conversion of x-datadog-* headers to tracestate for
+        representation in span links.
+        """
+        with test_library:
+            with test_library.start_span(
+                "root",
+                links=[Link(parent_id=0, attributes={"foo": "bar"})],
+                http_headers=[
+                    ["x-datadog-trace-id", "1234567890"],
+                    ["x-datadog-parent-id", "9876543210"],
+                    ["x-datadog-sampling-priority", "2"],
+                    ["x-datadog-origin", "synthetics"],
+                    ["x-datadog-tags", "_dd.p.dm=-4,_dd.p.tid=0000000000000010"],
+                ],
+            ):
+                pass
+
+        span = get_span(test_agent)
+        assert span.get("trace_id") != 1234567890
+        assert span_has_no_parent(span)
+        assert span["meta"].get(ORIGIN) is None
+        assert span["meta"].get("_dd.p.dm") != "-4"
+        assert span["metrics"].get(SAMPLING_PRIORITY_KEY) != 2
+
+        assert span.get("span_links") is not None
+
+        span_links = span["span_links"]
+        assert len(span_links) == 1
+        link = span_links[0]
+        assert link.get("span_id") == 9876543210
+        assert link.get("trace_id") == 1234567890
+        assert link.get("trace_id_high") == 16
+
+        assert link.get("tracestate") is not None
+        tracestateArr = link["tracestate"].split(",")
+        assert len(tracestateArr) == 1 and tracestateArr[0].startswith("dd=")
+        tracestateDD = tracestateArr[0][3:].split(";")
+        assert len(tracestateDD) == 3
+        assert "o:synthetics" in tracestateDD
+        assert "s:2" in tracestateDD
+        assert "t.dm:-4" in tracestateDD
+
+        assert (link.get("flags") or 0) == 0
+        assert len(link.get("attributes")) == 2
+        assert link["attributes"].get("foo") == "bar"
+        assert link["attributes"].get("_dd.p.dm") == "-4"
+
+    def test_span_link_from_distributed_w3c_headers(self, test_agent, test_library):
+        """Properly inject w3c distributed tracing information into span links.
+        This mostly tests that the injected tracestate and flags are accurate.
+        """
+        with test_library:
+            with test_library.start_span(
+                "root",
+                links=[Link(parent_id=0)],
+                http_headers=[
+                    ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                    ["tracestate", "foo=1,dd=t.dm:-4;s:2"],
+                ],
+            ):
+                pass
+
+        span = get_span(test_agent)
+        assert span.get("trace_id") != 1234567890
+        assert span_has_no_parent(span)
+        assert span["meta"].get("_dd.p.dm") != "-4"
```

Comment:
Now I'm a little confused. I thought we were merging the tracestate into the new local trace, but this is asserting that we do not. Can you help clarify this behavior?

Reply:
You are absolutely right and I messed this up.

Reply:
Ahah, I was confused on the same issue at line 101 😅

Reply:
Fixed now :-)